### ISSUE TITLE
Write initial draft of HPO GPT annotations manuscript

### ISSUE THREAD

**Issue by bschilder** [neurogenomics/RareDiseasePrioritisation#31]
I'll leave it up to you to decide how you'd like to write it up @KittyMurphy , but besides Google Docs this might be a good time to try writing a manuscript entirely in Rmarkdown or Quarto.

Existing code/results you can use as a basis for the manuscript:
- https://neurogenomics.github.io/RareDiseasePrioritisation/reports/HPO_annotations
- https://neurogenomics.github.io/RareDiseasePrioritisation/reports/gpt_annotations

**Comment by KittyMurphy** [comment]
Decided on quarto, markdown can be found here: https://github.com/neurogenomics/RareDiseasePrioritisation/tree/master/manuscript

**Comment by bschilder** [comment]
Just checking @KittyMurphy , is this the most up-to-date version of the GPT annotations?

https://github.com/neurogenomics/RareDiseasePrioritisation/blob/master/gpt_annotations/gpt4_hpo_annotations.csv

```R
hpo <- HPOExplorer::get_hpo()
path <- paste0(
      "https://github.com/neurogenomics/RareDiseasePrioritisation/raw/master/",
      "gpt_annotations/gpt4_hpo_annotations.csv"
    )
 d <- data.table::fread(path, header = TRUE)
length(unique(d$phenotype))
```
It currently contains 11159/18082 HPO phenotypes.

**Comment by KittyMurphy** [comment]
Thanks for the reminder, I've just uploaded the newest version @Schilder, Brian ***@***.***>
________________________________

**Comment by bschilder** [comment]
thanks! just downloaded it.

one thing im trying to sort out is, there's 917 terms that i cant match up to HPO IDs. these could either be:
- cases where the HPO term name changed over time within the HPO
- cases where GPT took some small liberties with renaming the HPO term

```r
path <- paste0(
      "https://github.com/neurogenomics/RareDiseasePrioritisation/raw/master/",
      "gpt_annotations/gpt4_hpo_annotations.csv"
    )
 d <- data.table::fread(path, header = TRUE)
 d <- HPOExplorer::add_hpo_id(d)
d[is.na(hpo_id)]
```

![Screenshot 2024-03-11 at 10 05 14](https://github.com/neurogenomics/RareDiseasePrioritisation/assets/34280215/b9dcb9a4-d066-4e74-be70-fe29f2fa63ef)

**Comment by KittyMurphy** [comment]
I just looked back at one of the prompts files and "3-hydroxyacyl-CoA dehydrogenase activity" and "White streaks/specks on enamel"  are both there, so at least for these phenotypes its a case of HPO term names changing.

**Comment by bschilder** [comment]
ok good to know! these changes should all be recorded somewhere by HPO, but working on figuring how to access deprecated IDs.
 
in the meantime, I may just try to use an old version of the HPO and see if that helps. do you know which version you used?

**Comment by KittyMurphy** [comment]
Re: HPO version it would be whatever was used in MultiEWCE and HPOExplorer. 

I originally got the HPO terms using: 

`all_res <- MultiEWCE::load_example_results()`

I then switched to: 
`all_res <- HPOExplorer::make_phenos_dataframe()`

Since February I've been using the  .obo file from the  [2024-02-08 Release](https://github.com/obophenotype/human-phenotype-ontology/releases/tag/v2024-02-08). If you follow the link you can see files associated with each release which might be useful in finding old/new terms!

**Comment by bschilder** [comment]
> Re: HPO version it would be whatever was used in MultiEWCE and HPOExplorer.

Right, but they don't use static versions anymore. they download them directly from the HPOs latest release at the time you first run the commands.

> Since February I've been using the .obo file from the [2024-02-08 Release](https://github.com/obophenotype/human-phenotype-ontology/releases/tag/v2024-02-08). If you follow the link you can see files associated with each release which might be useful in finding old/new terms!

Does this mean you used other versions too?

**Comment by bschilder** [comment]
Just implemented some changes in my mapping functions so that they can ignore case, which helped a bit but still have 851 terms with mismatched hpo names. Sharing this remaining list here so you can focus on just these @KittyMurphy 
[mismatched_hpo_names.csv.gz](https://github.com/neurogenomics/RareDiseasePrioritisation/files/14557972/mismatched_hpo_names.csv.gz)

**Comment by KittyMurphy** [comment]
So far, I've managed to get HPO ids for 723/851 terms. Sharing here: 
[mismatched_hpo_names_fixed.csv](https://github.com/neurogenomics/RareDiseasePrioritisation/files/14562614/mismatched_hpo_names_fixed.csv)

To achieve this, I used the HPO .obo file versions I used for the annotations (November 2023 and February 2024), either by directly merging on 'hpo_name', or by using the 'synonym' 

For ~300 terms I got the id using the HPO .obo versions I used for the annotations (November 2023 and February 2024). For the remainder I used the 'synonym' information, also from the .obo.
